### DISCUSSION TITLE
How many white spaces are recommended for tabs in markdown?

### DISCUSSION THREAD

**Discussion by mendax1234** [squidfunk/mkdocs-material#6026] [Q&A]
Hi, l want to use markdown to write something, and l use vscode as my editor. However, l find that vscode's default white spaces for tab in markdown is two. Should l change it to 4? And how to do that for markdown but not change the tab settings for other non-markdown files? 

Thanks for replying!

**Comment by facelessuser** [comment]
I would probably change it for Markdown at least. How you use tabs in other languages is really up to you. Conventions may vary depending on the community.

**Reply by alexvoss** [reply]
For Markdown you need at least one tab or 4 spaces for indenting pre-formatted code. Spaces also come into play when doing lists and because of the list symbol or numbering 4 spaces make more sense there as well. So, setting it to 4 would seem to make sense.

I searched the VS Code settings for indent and found "Editor: Detect Indentation", which turns on filetype specific settings. I could, however, not find a setting for Markdown - or JavaScript. It might be that selecting this just puts you at the mercy of what Microsoft collectively think is right. An alternative might be to use a Workspace-specific setting? 

Caveat: am only an occasional VS Code user. IMHO 4 spaces is good for all languages. Code that ends up being too deeply nested should probably be refactored. So, I

**Reply by squidfunk** [reply]
Another solution might be to create an `.editorconfig`, which allows you to define indent, line breaks, etc. for each language. It's a standard that all popular editors support and it ensures that your contributors use the same settings when creating PRs.

For reference, here's our [`.editorconfig`](https://github.com/squidfunk/mkdocs-material/blob/master/.editorconfig):

https://github.com/squidfunk/mkdocs-material/blob/9888723f86b20224197811fb647bfc4a79ce5a9f/.editorconfig#L21-L36

**Reply by roke-julian-lockhart** [reply]
> https://github.com/squidfunk/mkdocs-material/discussions/6026#discussioncomment-7022741

@alexvoss, https://github.com/github/markup/issues/1142#issuecomment-355010377 appears to demonstrate otherwise - that there merely needs to be a single character of whitespace, and that it needs to be consistent across lines. However, it's not well supported.

**Reply by squidfunk** [reply]
@RokeJulianLockhart GitHub uses CommonMark, while Python Markdown is an implementation of (the non-standard) Gruber Markdown, which mandates four spaces for lists, eight for sublists etc. This is a common pitfall among users, but we currently cannot just swap out Python Markdown with CommonMark, as it is closely intertwined with the MkDocs ecosystem.
